ape_etherscan/explorer.py

```diff
@@ -1,14 +1,32 @@
 from ape.api import ExplorerAPI
+from ape.types import ABI, ContractType
+from typing import List
+import requests
+import json
 
 ETHERSCAN_URI = (
     lambda n: (f"https://{n}.etherscan.io/" if n != "mainnet" else "https://etherscan.io/")
     + "{0}/{1}/"
 )
 
+ETHERSCAN_API_URI = lambda n: (
+    f"https://api-{n}.etherscan.io/api" if n != "mainnet" else "https://api.etherscan.io/api"
+)
+
 
 class Etherscan(ExplorerAPI):
     def get_address_url(self, address: str) -> str:
         return ETHERSCAN_URI(self.network.name).format("address", address)
 
     def get_transaction_url(self, transaction_hash: str) -> str:
         return ETHERSCAN_URI(self.network.name).format("tx", transaction_hash)
+
+    def get_contract_abi(self, address: str) -> List[ABI]:
+        response = requests.get(
```

Comment:
Make sure to call `response.raise_for_status()`.
Otherwise 400, 403, etc won't raise errors and the program will likely crash somewhere else in a more confusing spot

Reply:
ah ok thank you i'll give that a shot

Reply:
https://github.com/ApeWorX/ape-etherscan/issues/6